docs/source/killed.rst

```diff
@@ -0,0 +1,142 @@
+.. _killed:
+
+Why did my worker die?
+----------------------
+
+A Dask worker can cease functioning for a number of reasons. These fall into the
+following categories:
+
+- the worker chooses to exit
+- an unrecoverable exception happens within the worker
+- the worker process is shut down by some external action
+
+Each of these cases will be described in more detail below. The *symptoms* you will
+experience when these things happen range from simply work not getting done anymore,
+to various exceptions appearing when you interact with your local client, such as
+``KilledWorker``, ``TimeoutError`` and ``CommClosedError``.
+
+Note the special case of ``KilledWorker``: this means that a particular task was
```

Comment:
The Dask scheduler does not differentiate between different task states on a worker. So, if there are 1000 tasks on a worker and only one is executed right now, then a segfault or out of memory will still lead to all 1000 tasks being suspected and having their error counter incremented. 

In practice this can lead to situations where an unlucky but totally innocent tasks first crosses the `allowed-failure` threshold even though other tasks are actually responsible. If you then try to investigate the `KilledWorker` exception you end up looking at the wrong task

Even though this is an implementation detail, this can make debugging quite difficult and might be worth mentioning here.

Reply:
I am surprised - does the worker not notify the scheduler which task it's about to process?